### ISSUE TITLE
[Question] Import-blacklist by regex does not work as expected 

### ISSUE THREAD

**Issue by alxpsr** [palantir/tslint#4660]
### Question

- __TSLint version__: 5.15.0
- __TypeScript version__: 3.4.3
- __Running TSLint via__: CLI / VSCode 

#### TypeScript code being linted

```ts
import { SecondaryService } from "./src";
export class BasicService {
    public State: boolean;

    constructor(public s: SecondaryService) {

    }
}
```

with `tslint.json` configuration:

```
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended"
    ],
    "jsRules": {},
    "rules": {
        "import-blacklist": [
            true,
            "^.*src$",
        ]
    },
    "rulesDirectory": []
}
```

#### Actual behavior
Tslint does not emit any errors

#### Expected behavior
string `import { SecondaryService } from "./src";` should mark as wrong by import-blacklist regex rule

#### Mini-repo
https://github.com/alxpsr/tslint-regex

#### Questions
- What im doing wrong? 
- How can i disallow importing from `import-blacklist` regex patterns?

**Comment by JoshuaKGoldberg** [comment]
Solution: wrap any regular expressions in arrays. ` ["^.*src$"]` instead of `"^.*src$"`.

This isn't described well in the documentation; accepting PRs to explicitly mention this!

```json
{
    "defaultSeverity": "error",
    "jsRules": {},
    "rules": {
        "import-blacklist": [
            true,
            ["^.*src$"]
        ]
    },
    "rulesDirectory": []
}
```

**Comment by alxpsr** [comment]
Thanks

**Comment by alxpsr** [comment]
When i run `tslint --project .` tslint no emit any error. But in VsCode my import mark as green (probably as warning). On the other hand if i added `rxjs` to config and to my `src/service.ts` - tslint emit error (VsCode anyway mark import as warning). Why do the two strategies (regex-pattern vs simple string) have different behaviors?

**Comment by alxpsr** [comment]
Guys?

**Comment by JoshuaKGoldberg** [comment]
Hi @alxpsr - please file a separate issue. It's better to have one issue per discussion (and one discussion per issue) to keep these issues organized. Differences in lint complaints in VS Code vs the CLI are a separate issue from the documentation here.

**Comment by JoshuaKGoldberg** [comment]
Sorry for the ambiguity - the original issue here is still accepting PRs 😄 just there should be a second issue.

**Comment by alshdavid** [comment]
```json
{
    "defaultSeverity": "error",
    "jsRules": {},
    "rules": {
        "import-blacklist": [
            true,
            ["^.*src$"]
        ]
    },
    "rulesDirectory": []
}
```

doesn't match on organisation names like `@mylib/thing/src`

**Comment by sandrooco** [comment]
@alshdavid Could you find a solution by now? I have exactly the same issue right now with ["^.*@app/bkh.*"].

**Comment by alshdavid** [comment]
No solution, sadly

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
